### PR TITLE
ref(pkg/chartutil): replace use of 'fmt.Efforf'

### PR DESCRIPTION
**Pull request by rbwsam**
'fmt.Errorf' is unnecessary when checking for 0 length path in Values.PathValue
due to the lack of formatting. Add test covering changes.

### CODE REVIEW

**Comment by rbwsam** [comment]
Commenting to trigger check of CLA.

**Comment by rbwsam** [comment]
Triggering CLA check again.

**Review by bacongobbler — APPROVED** [review]

File: pkg/chartutil/values.go [hunk 1]
```diff
@@ -390,7 +391,7 @@ func istable(v interface{}) bool {
 // PathValue takes a yaml path with . notation and returns the value if exists
 func (v Values) PathValue(ypath string) (interface{}, error) {
 	if len(ypath) == 0 {
-		return nil, fmt.Errorf("yaml path string cannot be zero length")
+		return nil, errors.New("yaml path string cannot be zero length")
```
Comment (seh) [inline pkg/chartutil/values.go]: - [ ] While we're here, consider capitalizing "YAML" in the error message.

File: pkg/chartutil/values_test.go [hunk 2]
```diff
@@ -447,10 +447,12 @@ chapter:
 	if _, err := d.PathValue("chapter.doesntexist.one"); err == nil {
 		t.Errorf("Non-existent key in middle of path should return error: %s\n%v", err, d)
 	}
+	if _, err := d.PathValue(""); err == nil {
+		t.Errorf("Empty path should return error: %s\n%v", err, d)
```
Comment (seh) [inline pkg/chartutil/values_test.go]: - [ ] Since the `err` value is known to be nil, don't include it in your complaint.

**Review by seh — CHANGES_REQUESTED** [review]

File: pkg/chartutil/values_test.go (hunk 2 above)
Comment (rbwsam) [inline pkg/chartutil/values_test.go]: Should I even include the document in the complaint? Seems kind of irrelevant in this case.

Comment (seh) [inline pkg/chartutil/values_test.go]: I would replace the statement with the following:
```go
t.Error("Asking for the value from an empty path should yield an error")
```
I didn't punctuate the sentence for consistency with the rest of the failure messages in this file.

Comment (rbwsam) [inline pkg/chartutil/values_test.go]: Thanks!

File: pkg/chartutil/values.go [hunk 3]
```diff
@@ -390,7 +391,7 @@ func istable(v interface{}) bool {
 // PathValue takes a yaml path with . notation and returns the value if exists
```
Comment (seh) [inline pkg/chartutil/values.go]: - [ ] Likewise, capitalize "YAML" here too, since you're covering this function in this patch.  
  It's not clear what a "YAML path" is, though. I think this comment is referring to something akin to field selector syntax applied to values read from a YAML file.

**Comment by rbwsam** [comment]
@seh I tried to improve the comment, what do you think?

File: pkg/chartutil/values.go [hunk 4]
```diff
@@ -388,7 +388,13 @@ func istable(v interface{}) bool {
 	return ok
 }
 
-// PathValue takes a yaml path with . notation and returns the value if exists
+// PathValue takes a path that traverses a YAML structure and returns the value at the end of that path.
+// The path starts at the root of the YAML structure and is comprised of YAML keys separated by periods.
+// Given the following YAML data the value at path "chapter.one.title" is "Loomings".
+//
+// 		chapter:
```
Comment (seh) [inline pkg/chartutil/values.go]: The comment is excellent. My only hesitation is the generous indentation of the example document fragment. It's hard to tell reading it in GitHub, but it looks like a space and two tabs. I'm not sure how much indentation it takes to get Godoc to render the fragment as pre-formatted text.

**Comment by rbwsam** [comment]
@seh You are correct, it was a space and two tabs. I changed it to a single tab and generated the doc with `godoc` to confirm that it is rendered as pre-formatted.

**Review by thomastaylor312 — APPROVED** [review]
This looks good! Thank you for adding the extra test too. Because @seh was the original reviewer, I am going to let him merge.

**Review by seh — APPROVED** [review]
